application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/FuzzySearchSuccessTest.java

```diff
@@ -0,0 +1,163 @@
+   * Given valid search criteria terms, ensure the API returns a 200 success code, and a list of
+   * services are returned.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void oneSearchCriteriaGiven() throws Exception {
+    HttpEntity<String> request = new HttpEntity<String>(null, headers);
+    UriComponentsBuilder uriBuilder =
+        UriComponentsBuilder.fromHttpUrl(endpointUrl).queryParam("search_criteria", "Term1");
+    ResponseEntity<String> response =
+        restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, request, String.class);
+
+    assertTrue(response.getStatusCode() == HttpStatus.OK);
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
+
+    assertEquals(apiSuccResponse.getNumberOfServices(), 2);
+
+    // Add the service ids that you want to verify
+    List<Integer> serviceIdsToVerify = new ArrayList<>();
+    serviceIdsToVerify.add(1);
+    serviceIdsToVerify.add(2);
+
+    verifySuccessResponse(apiSuccResponse, serviceIdsToVerify);
+  }
+
+  /**
+   * Given at least one valid search term, ensure the API returns a 200 success code, and a list of
+   * services are returned.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void atLeastOneValidSearchTermGiven() throws Exception {
+    HttpEntity<String> request = new HttpEntity<String>(null, headers);
+    UriComponentsBuilder uriBuilder =
+        UriComponentsBuilder.fromHttpUrl(endpointUrl)
+            .queryParam("search_criteria", "a")
+            .queryParam("search_criteria", "ab")
+            .queryParam("search_criteria", "Term1");
+    ResponseEntity<String> response =
+        restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, request, String.class);
+
+    assertTrue(response.getStatusCode() == HttpStatus.OK);
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
+
+    assertEquals(apiSuccResponse.getNumberOfServices(), 2);
+
+    // Add the service ids that you want to verify
+    List<Integer> serviceIdsToVerify = new ArrayList<>();
+    serviceIdsToVerify.add(1);
+    serviceIdsToVerify.add(2);
+
+    verifySuccessResponse(apiSuccResponse, serviceIdsToVerify);
+  }
+
+  /**
+   * Given valid search criteria terms, ensure the API returns a 200 success code, and a list of no
+   * services are returned.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void oneSearchCriteriaGivenNoResults() throws Exception {
+    HttpEntity<String> request = new HttpEntity<String>(null, headers);
+    UriComponentsBuilder uriBuilder =
+        UriComponentsBuilder.fromHttpUrl(endpointUrl).queryParam("search_criteria", "Term0");
+    ResponseEntity<String> response =
+        restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, request, String.class);
+
+    assertTrue(response.getStatusCode() == HttpStatus.OK);
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
```

Comment:
Can we use an expressive name for this class e.g. `ApiSuccessResponse` rather than an abbreviation?

Reply:
Agreed.